### PR TITLE
Update cypress to run on firefox

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link: -

### Problem Statement

Our environment on the phone is firefox browser

### Solution

Update the cypress configuration and our app feature flag to be able to run on firefox

### Note

_I believe our app won't be able to work correctly when KaiOS updates the gecko engine to the latest, for example, if you run our app on chrome, the article page navigation doesn't work well due to the reason we are now developing it for the older firefox. The good thing is we may not need the `isProd` utils for setting request header._

### CODE REVIEW

File: cypress/integration/article-view-spec.js [hunk 1]
```diff
@@ -177,7 +177,9 @@ describe('Article view', () => {
   it('check text size remains after changing on preview', () => {
     goToCatArticle()
     articlePage.getArticleText().should('have.attr', 'style', 'font-size: 16px;')
-    cy.downArrow().downArrow().downArrow().enter()
+    articlePage.selectOptionFromActionsMenu('sections')
```
Comment (jpita) [inline cypress/integration/article-view-spec.js:180]: you found the issue? awesome!
I was confused because it was working fine on my local...

**Comment by jpita** [comment]
how can we know the firefox version installed on the CI?

File: cypress/integration/article-view-spec.js (hunk 1 above)
Comment (hueitan) [reply cypress/integration/article-view-spec.js:180]: I was confused too

Comment (hueitan) [reply cypress/integration/article-view-spec.js:180]: chrome works, but the user sees the page as blank, but able to select the preview.
but i'm not sure how it looks like in the terminal cypress

**Comment by jpita** [comment]
> how can we know the firefox version installed on the CI?

I think we can add 
`navigator.appVersion` 
in a 
`console.log`
in a test

**Comment by jpita** [comment]
> > how can we know the firefox version installed on the CI?
> 
> I think we can add
> `navigator.userAgent`
> in a
> `console.log`
> in a test
no need for console.log, cypress prints it out
image

File: cypress/integration/article-view-spec.js (hunk 1 above)
Comment (hueitan) [reply cypress/integration/article-view-spec.js:180]: image

hmmm when checking the video, it opens the reference preview not article preview ...

Comment (hueitan) [reply cypress/integration/article-view-spec.js]: Ok, I have a slightly changed to the behaviour here, due to the window.screen reason, we don't know where it goes when press down 3 times, therefore press 1 time to get more predictable location either next page or next section.

File: .circleci/config.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
           name: Build and Test
           command: |
             npm install
-            npm run build
+            node ./node_modules/cross-env/src/bin/cross-env.js DISABLE_REQUEST_HEADER=1 npm run build
```
Comment (stephanebisson) [inline .circleci/config.yml:12]: Please encapsulate that as a command in package.json

File: src/utils/cachedFetch.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ export const cachedFetch = (url, transformFn, cache = true) => {
   const promise = new Promise((resolve, reject) => {
     xhr.responseType = 'json'
     xhr.open('GET', url)
-    if (isProd()) {
+    if (isProd() && !isRequestHeaderDisabled()) {
```
Comment (stephanebisson) [inline src/utils/cachedFetch.js:18]: Does it make sense to have both conditions here?

**Comment by hueitan** [comment]
> I'm not really convinced of this change, but I'm also not against it. Have we seen situations where new Firefox behaves more similar to old Firefox than Chrome does?

That's what my initial thought, our app runs on the older firefox browser. we can't use the exact same environment as the device in the cypress test. According to what @jpita found, page flip works in firefox but not chrome (that I can confirm on chrome), can you confirm this again?

> Also, I don't have new Firefox locally, will the tests still work with Chrome?

Yes, the test still works in Chrome.

I also don't have any strong opinion on this, running both chrome/firefox can give us an overview of the test spec, which both are awesome!

File: src/utils/cachedFetch.js (hunk 3 above)
Comment (hueitan) [reply src/utils/cachedFetch.js:18]: agree with having only one condition here.

**Comment by hueitan** [comment]
@jpita ready to be reviewed, please test local and app carefully.

**Comment by jpita** [comment]
@hueitan can u fix the conflicts when u have some time please?

**Comment by hueitan** [comment]
Done @jpita

**Pull request merged**